### ISSUE TITLE
Outdated docs page for Anatomy of a Type

### ISSUE THREAD

**Issue by SteveJobzniak** [phpDocumentor/phpDocumentor2#1896]
http://docs.phpdoc.org/guides/types.html lists `$this` as a valid Type keyword.

But that page links to the "definition of a Type" page (http://docs.phpdoc.org/references/phpdoc/types.html) which says that `$this` is not a valid type.

**Comment by jaapio** [comment]
Technically $this is not a keyword neither a type. But we do support it in the docblocks since many users are using it as an alias of self or static.

**Comment by SteveJobzniak** [comment]
Yep, it's a great way to say `$this`: "returns THE current instance" rather than `self`: "returns AN instance (maybe a new one, who knows...)".

**Comment by caramelomartins** [comment]
@jaapio If it is neither a keyword nor a type, do you want to remove the reference from http://docs.phpdoc.org/guides/types.html? Or, since there's support for it, do you wish to change the reference page?

Or neither?

**Comment by SteveJobzniak** [comment]
@caramelomartins The `$this` keyword is used by lots of IDEs to distinguish between `$this` (THIS instance) and `self` (AN instance). And @jaapio said "But we do support it (`$this`) in the docblocks since many users are using it as an alias of self or static."

It just seems like it's not an officially registered keyword in PHPdoc.

It should be. It's important to be able to identify when setters are returning `$this` so that they can be chained.

**Comment by jaapio** [comment]
The $this notation is not resolved in all situations to the correct class. The only valid situation to use it is in the @return annotation.

Personally I never use this notation. But that's not the statement of the project. We want to support what our users expect to be possible.

I think it should be removed from the types page. And added to the return tag docs to explain the context. And why you might want to use it.

**Comment by SteveJobzniak** [comment]
Ahhh. Yep I only use it in @return. It makes no sense in other places.

In that case I definitely agree that it should be removed from types and added to @return docs.
